__tests__/packages.ts

```diff
@@ -0,0 +1,120 @@
+import { matchersWithOptions } from "jest-json-schema";
+import glob from "glob";
+import fs from "fs";
+import path from "path";
+import { allPackageFiles } from "../tools/shared/packages";
+import childProcess from "child_process";
+
+describe("monorepo content validation", () => {
+  test("the monorepo does not contain banned words", async () => {
+    const checkBannedWords = await new Promise<void>((res, reject) => {
+      childProcess.exec(
+        "./wordCheck",
+        { cwd: path.resolve(__dirname) },
+        (error, _stuff, stderr) => {
+          if (error || stderr) {
+            return reject(error || stderr);
+          }
+          return res();
+        }
+      );
+    });
+    expect(checkBannedWords).resolves;
+  });
+});
+
+describe("package.json validation", () => {
+  const packageFiles: string[] = allPackageFiles(glob);
+  const genericSchema = require("./schemaGeneric.json");
+  const publicSchema = require("./schemaPublic.json");
+  const restrictedSchema = require("./schemaRestricted");
+
+  expect.extend(
+    matchersWithOptions({
+      schemas: [genericSchema, publicSchema, restrictedSchema],
+    })
+  );
+
+  //used for license checker, but should be generated ahead of time with a fully populated list
+  const excludedPackages = [
+    "@newrelic/aws-sdk@1.1.3",
+    "@newrelic/koa@3.0.0",
+    "@newrelic/native-metrics@5.1.0",
+    "@newrelic/native-metrics@5.2.0",
+    "@newrelic/superagent@2.0.1",
+    "emitter-component@1.1.1",
+    "facebook-nodejs-business-sdk@12.0.1",
+  ];
+
+  packageFiles.forEach((pkgFile) => {
+    const pkgJson = JSON.parse(fs.readFileSync(pkgFile, "utf-8"));
+    const nameAndVersion = `${pkgJson.name}@${pkgJson.version}`;
+    excludedPackages.push(nameAndVersion);
+  });
+
+  const pkgFile =
+    "/Users/teal/workspace/grouparoo/grouparoo/plugins/@grouparoo/customerio/package.json";
+  describe.each(packageFiles)(`%s`, (pkgFile) => {
```

Comment:
I would not have expected this to pass on CI - there's no `/teal` folder there!  This should be fixed, but also... why did it pass?

Reply:
Ah.  This was left over from debugging, oops.... it's still passing because the `pkgFile` referenced below is from the loop, not the one declared up here.

Reply:
Ah! Then 👍 once this is removed!